lib/puppet/provider/cisco_interface_portchannel/nxapi.rb

```diff
@@ -0,0 +1,159 @@
+#
+# The NXAPI provider for cisco_interface_portchannel
+#
+# Dec 2015
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+begin
+  require 'puppet_x/cisco/autogen'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..',
+                                     'puppet_x', 'cisco', 'autogen.rb'))
+end
+
+Puppet::Type.type(:cisco_interface_portchannel).provide(:nxapi) do
+  desc 'The NXAPI provider for cisco_interface_portchannel'
+
+  confine feature: :cisco_node_utils
+  defaultfor operatingsystem: :nexus
+
+  mk_resource_methods
+
+  INTF_PC_NON_BOOL_PROPS = [
+    :lacp_max_bundle,
+    :lacp_min_links,
+    :port_hash_distribution,
+  ]
+  INTF_PC_BOOL_PROPS = [
+    :lacp_graceful_convergence,
+    :lacp_suspend_individual,
+    :port_load_defer,
+  ]
+  INTF_PC_ALL_PROPS = INTF_PC_NON_BOOL_PROPS + INTF_PC_BOOL_PROPS
+
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self,
+                                            '@interface_portchannel',
```

Comment:
Chris recently used a different convention when referring to the `node_utils` object that I really like and makes the code more readable.  It's a simple query/replace but makes it clear that we are referring to the node_utils object for this provider.  He and I agreed that we want to do this moving forward for all new providers.

1) Change `@interface_portchannel` references to just `@nu`.
2) Change `intf` references in `self.properties_get` and `self.instances` to `nu_obj`

Here is a pointer to his code for reference.

https://github.com/cisco/cisco-network-puppet-module/blob/develop/lib/puppet/provider/cisco_vdc/nxapi.rb